src/ddsrt/src/xmlparser.c

```diff
@@ -377,22 +377,27 @@ static void discard_payload (struct ddsrt_xmlp_state *st)
   st->tpescp = 0;
 }
 
-static int append_to_payload (struct ddsrt_xmlp_state *st, int c, int islit)
+static void append_literal_to_payload (struct ddsrt_xmlp_state *st, int c)
```

Comment:
I suppose, for the changes in this file the only issue I have is the same as for the proposed change in `hopscotch.c`.

Reply:
I'll swap this out to using a `dds_retcode_t` and `ddsrt_realloc_s`. Before I make that change would you like me to revert the changes so that `append_to_payload` and `append_literal_to_payload` are unified again in `append_to_payload`?

Reply:
No, there is something to be said for having two different functions there, especially with `append_to_payload` defined in terms of `append_literal_to_payload`

Reply:
Okay, I kept the two functions separate and made `append_to_payload` return an error in case the reallocation fails (+reintroduced the previous error checking).